### PR TITLE
Mnemonic support

### PR DESCRIPTION
**Pull request by aguycalled**
This PR adds a new RPC command to export the existing master private key encoded as a mnemonic:

dumpmnemonic

It admits an argument specifying the language.

Support for two new wallet options (-importmnemonic and -mnemoniclanguage) have also been added to allow to create a new wallet from the specified mnemonic.

Testing needed.

### CODE REVIEW

**Comment by matt-auckland** [comment]
cACK, will test this when I have spare time

File: src/mnemonic/mnemonic.cpp [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        {
+            words.push_back(sentence.substr(prevPos, i - prevPos));
+            prevPos = i + 1;
+        }
+    }
+    return words;
+}
+
+
+//long_hash decode_mnemonic(const word_list& mnemonic,
```
Comment (matt-auckland) [inline src/mnemonic/mnemonic.cpp]: Is there a reason we need to leave this code in?

File: src/qt/navcoingui.cpp [hunk 2]
```diff
@@ -150,13 +150,13 @@ NavCoinGUI::NavCoinGUI(const PlatformStyle *platformStyle, const NetworkStyle *n
     prevBlocks(0),
     spinnerFrame(0),
     fDontShowAgain(false),
-    lastDialogShown(0),
     unlockWalletAction(0),
     lockWalletAction(0),
     toggleStakingAction(0),
```
Comment (matt-auckland) [inline src/qt/navcoingui.cpp]: What is this code for?

File: src/qt/navtechinit.cpp [hunk 3]
```diff
@@ -25,7 +25,6 @@ NavTechInit::~NavTechInit()
 void NavTechInit::ShowNavtechIntro(bool exitAfter)
 {
     NavTechInit navtechinit;
```
Comment (matt-auckland) [inline src/qt/navtechinit.cpp]: What is this code for?

**Comment by craigmacgregor** [comment]
@matt-auckland going to evaluate functionality and report back

**Comment by marcus290** [comment]
I manually tested `dumpmnemonic` in different languages and `-importmnemonic` and `-mnemoniclanguage` and it works for me.

Just wondering, why not add support for the remaining languages that are already in dictionary.cpp?

I also think 80512622b488ee2d232b797ace130fea6b05d15e maybe should be moved out to a separate commit.

**Pull request merged**